MDANSE_GUI/Src/MDANSE_GUI/Tabs/Visualisers/TextInfo.py

```diff
@@ -47,3 +49,53 @@ def filter(self, some_text: str, line_break="<br />"):
         if self._footer:
             new_text += line_break + self._footer
         return new_text
+
+
+class MathInfo(TextInfo):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @staticmethod
+    def scan(text: str) -> list[tuple[str, bool]]:
+        # Instantiate renderer object
+        renderer = MathRenderer(text)
+
+        # Scan text for existence of raw LaTex expressions
+        scanned = renderer.scan()
+
+        # Iterate over scanned text, rendering LaTex substrings if image not already cached
+        for token, is_expression in scanned:
+            if is_expression and not MathRenderer.cached(token):
+                renderer.render(
+                    token,
+                    QApplication.styleHints().colorScheme() == Qt.ColorScheme.Dark,
+                )
+
+        return scanned
+
+    def filter(self, some_text: str, line_break="<br />"):
+        filtered = super().filter(some_text, line_break)
+        scanned = self.scan(filtered)
+
+        html_substrings = []
+        for token, is_expression in scanned:
+            if is_expression:
+                image = MathRenderer.from_cache(token)
+                if len(token) < 10:
```

Comment:
Bear in mind that latex tokens may not correspond to large images:

`\underbrace{\mathbf{x}} = \mathbf{3}`

$\underbrace{\mathbf{x}} = \mathbf{3}$